transaction/src/manifest/generator.rs

```diff
@@ -779,6 +843,42 @@ fn generate_resource_address(
     }
 }
 
+fn generate_dynamic_global_address(
```

Comment:
I'm surprised we don't see equivalents of these for eg `generate_dynamic_resource_address` for eg creating proofs or buckets? Which either resolves a resource address or a named address?

Or is this by design so we can statically resolve these? (Although `deposit_batch` suggests we don't care about this too much!)

(And ditto with `generate_dynamic_package_address`)

Reply:
Right, I explained this a bit in `instruction.rs`.